### PR TITLE
GetPaths & x86

### PR DESCRIPTION
**Pull request by Julien-Molina**

### CODE REVIEW

File: Cake.XComponent.Test/Cake.XComponent.Test.csproj [hunk 1]
```diff
@@ -45,14 +45,21 @@
     </Reference>
     <Reference Include="System" />
     <Reference Include="System.Core" />
+    <Reference Include="System.Data" />
+    <Reference Include="System.Drawing" />
```
Comment (maurelio1234) [inline Cake.XComponent.Test/Cake.XComponent.Test.csproj]: Drawing? :O

File: Cake.XComponent.Test/XcPathAliasesTest.cs [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+            Assert.IsNull(cakeContext.GetXcStudioPath(Platform.X86));
+        }
+
+        [TestCase(Platform.X64)]
+        [TestCase(Platform.X86)]
+        public void TestGetXcBuildPath(Platform platform)
+        {
+            WriteResource("Cake.XComponent.Test.Input.Cake.XComponent.Test.FakeExe.exe", _applicationDirectory, PathFinder.GetXcBuildProgram(platform));
+            var cakeContext = Substitute.For<ICakeContext>();
+            cakeContext.GetXcBuildPath(platform);
```
Comment (maurelio1234) [inline Cake.XComponent.Test/XcPathAliasesTest.cs]: what's the point?

File: Cake.XComponent.Test/XcPathAliasesTest.cs [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+            Assert.IsNull(cakeContext.GetXcBuildPath(Platform.X86));
+        }
+
+        [TestCase(Platform.X64)]
+        [TestCase(Platform.X86)]
+        public void TestGetXcRuntimePath(Platform platform)
+        {
+            WriteResource("Cake.XComponent.Test.Input.Cake.XComponent.Test.FakeExe.exe", _applicationDirectory, PathFinder.GetXcRuntimeProgram(platform));
+            var cakeContext = Substitute.For<ICakeContext>();
+            cakeContext.GetXcRuntimePath(platform);
```
Comment (maurelio1234) [inline Cake.XComponent.Test/XcPathAliasesTest.cs]: What's the point?

File: Cake.XComponent.Test/XcPathAliasesTest.cs [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+        }
+
+        [Test]
+        public void TestSetXcRuntimePathWithNotExistingPath()
+        {
+            var path = Path.Combine(_applicationDirectory, PathFinder.GetXcRuntimeProgram(Platform.X64));
+            var cakeContext = Substitute.For<ICakeContext>();
+            cakeContext.SetXcRuntimePath(path);
+
+            Assert.IsNull(cakeContext.GetXcRuntimePath(Platform.X64));
```
Comment (maurelio1234) [inline Cake.XComponent.Test/XcPathAliasesTest.cs]: Why?

File: Cake.XComponent.Test/XcPathAliasesTest.cs [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+            Assert.IsNull(cakeContext.GetXcRuntimePath(Platform.X86));
+        }
+
+        [TestCase(Platform.X64)]
+        [TestCase(Platform.X86)]
+        public void TestGetXcBridgePath(Platform platform)
+        {
+            WriteResource("Cake.XComponent.Test.Input.Cake.XComponent.Test.FakeExe.exe", _applicationDirectory, PathFinder.GetXcBridgeProgram(platform));
+            var cakeContext = Substitute.For<ICakeContext>();
+            cakeContext.GetXcBridgePath(platform);
```
Comment (maurelio1234) [inline Cake.XComponent.Test/XcPathAliasesTest.cs]: What's the point?

File: Cake.XComponent.Test/XcPathAliasesTest.cs [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+        }
+
+        [Test]
+        public void TestSetXcBridgePathWithNotExistingPath()
+        {
+            var path = Path.Combine(_applicationDirectory, PathFinder.GetXcBridgeProgram(Platform.X64));
+            var cakeContext = Substitute.For<ICakeContext>();
+            cakeContext.SetXcBridgePath(path);
+
+            Assert.IsNull(cakeContext.GetXcBridgePath(Platform.X64));
```
Comment (maurelio1234) [inline Cake.XComponent.Test/XcPathAliasesTest.cs]: Why?

File: Cake.XComponent.Test/XcPathAliasesTest.cs [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+            Assert.IsNull(cakeContext.GetXcBridgePath(Platform.X86));
+        }
+
+        [TestCase(Platform.X64)]
+        [TestCase(Platform.X86)]
+        public void TestGetXcSpyPath(Platform platform)
+        {
+            WriteResource("Cake.XComponent.Test.Input.Cake.XComponent.Test.FakeExe.exe", _applicationDirectory, PathFinder.GetXcSpyProgram(platform));
+            var cakeContext = Substitute.For<ICakeContext>();
+            cakeContext.GetXcSpyPath(platform);
```
Comment (maurelio1234) [inline Cake.XComponent.Test/XcPathAliasesTest.cs]: What's the point?

File: Cake.XComponent.Test/XcPathAliasesTest.cs [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+        }
+
+        [Test]
+        public void TestSetXcSpyPathWithNotExistingPath()
+        {
+            var path = Path.Combine(_applicationDirectory, PathFinder.GetXcSpyProgram(Platform.X64));
+            var cakeContext = Substitute.For<ICakeContext>();
+            cakeContext.SetXcSpyPath(path);
+
+            Assert.IsNull(cakeContext.GetXcSpyPath(Platform.X64));
```
Comment (maurelio1234) [inline Cake.XComponent.Test/XcPathAliasesTest.cs]: Why?

File: Cake.XComponent/Utils/PathFinder.cs [hunk 9]
```diff
@@ -8,8 +8,12 @@ namespace Cake.XComponent.Utils
 {
     internal class PathFinder
     {
-        internal const string XcStudioExe = "XCStudio.exe";
-        internal const string XcBuildExe = "xcbuild.exe";
+        private const string XcStudioProgram = "XCStudio{0}.exe";
+        private const string XcSpyProgram = "xcspy{0}.exe";
```
Comment (maurelio1234) [inline Cake.XComponent/Utils/PathFinder.cs]: That's an inconsistency to fix (eventually) on the executable... sometimes it's XC, sometimes xc

File: Cake.XComponent/Utils/PathFinder.cs [hunk 10]
```diff
@@ -8,8 +8,12 @@ namespace Cake.XComponent.Utils
...
+        private const string XcRuntimeProgram = "xcruntime{0}.exe";
+        private const string XcBridgeProgram = "XCWebSocketBridge{0}.exe";
+        private const string XcBuildProgram = "xcbuild{0}.exe";
```
Comment (maurelio1234) [inline Cake.XComponent/Utils/PathFinder.cs]: some times it's lower case, sometimes camel case...

File: Cake.XComponent/Utils/PathFinder.cs [hunk 11]
```diff
@@ -8,8 +8,12 @@ namespace Cake.XComponent.Utils
...
+        private const string X86SubName = "32";
```
Comment (anouarhassine) [inline Cake.XComponent/Utils/PathFinder.cs]: Minor: I would call this _X86Suffix_

**Pull request merged**